nav2_behaviors/include/nav2_behaviors/timed_behavior.hpp

```diff
@@ -90,7 +90,7 @@ class TimedBehavior : public nav2_core::Behavior
   // Implement the behavior such that it runs some unit of work on each call
   // and provides a status. The Behavior will finish once SUCCEEDED is returned
   // It's up to the derived class to define the final commanded velocity.
-  virtual ResultStatus onCycleUpdate() = 0;
+  virtual ResultStatus onCycleUpdate(std::shared_ptr<typename ActionT::Result> result) = 0;
```

Comment:
Wouldn't this make alot more sense in `onActionCompletion`?

Reply:
Please do. I dont think cycle updates should need to populate the result. The ‘action complete’ callback seems like the right place